### PR TITLE
Implemented HSMBookKeepingClientProvider

### PR DESCRIPTION
**Pull request by kelvinator07**

### CODE REVIEW

File: src/test/java/co/rsk/federate/signing/hsm/advanceBlockchain/HSMBookKeepingClientProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Created by Kelvin Isievwore on 27/03/2023.
+ */
+public class HSMBookKeepingClientProviderTest {
+
+    private HSMBookKeepingClientProvider hsmBookKeepingClientProvider;
+    private HSMClientProtocol hsmClientProtocol;
```
Comment (nathanieliov) [inline src/test/java/co/rsk/federate/signing/hsm/advanceBlockchain/HSMBookKeepingClientProviderTest.java:20]:
```suggestion
    private HSMClientProtocol hsmClientProtocol;
```

Also, correct this `HSM Client Protocol`, I believe it should be named using camel case no matter "HSM" is an acronym. In fact, you used camel case when naming the implementation `HsmBookkeepingClientImpl`. So, let's be consistent with the naming.

Comment (kelvinator07) [reply src/test/java/co/rsk/federate/signing/hsm/advanceBlockchain/HSMBookKeepingClientProviderTest.java:20]: I Don't Understand Your Suggestion.. Elaborate Further?

File: src/main/java/co/rsk/federate/signing/hsm/advanceBlockchain/HSMBookKeepingClientProvider.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    private static final Logger logger = LoggerFactory.getLogger(HSMBookKeepingClientProvider.class);
+    private static final int MIN_SUPPORTED_VERSION = 2;
+    private static final int MAX_SUPPORTED_VERSION = 3;
+    private final HSMClientProtocol hsmClientProtocol;
+
+    public HSMBookKeepingClientProvider(HSMClientProtocol protocol) {
+        this.hsmClientProtocol = protocol;
+    }
+
+    public HSMBookkeepingClient getHSMBookkeepingClient() throws HSMClientException {
```
[the remark this answers is not in the document]
Comment (kelvinator07) [reply src/main/java/co/rsk/federate/signing/hsm/advanceBlockchain/HSMBookKeepingClientProvider.java]: Resolved

File: src/test/java/co/rsk/federate/signing/hsm/advanceBlockchain/HSMBookKeepingClientProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    public void getHSMBookkeepingClient_v2() throws HSMClientException {
+        when(hsmClientProtocol.getVersion()).thenReturn(2);
+        HSMBookkeepingClient bookkeepingClient = hsmBookKeepingClientProvider.getHSMBookkeepingClient();
+
+        Assert.assertTrue(bookkeepingClient instanceof HsmBookkeepingClientImpl);
+    }
+
+    @Test()
+    public void getHSMBookkeepingClient_v3() throws HSMClientException {
+        when(hsmClientProtocol.getVersion()).thenReturn(2);
```
[the remark this answers is not in the document]
Comment (kelvinator07) [reply src/test/java/co/rsk/federate/signing/hsm/advanceBlockchain/HSMBookKeepingClientProviderTest.java]: Resolved

File: src/test/java/co/rsk/federate/signing/hsm/advanceBlockchain/HSMBookKeepingClientProviderTest.java (hunk 1 above)
Comment (nathanieliov) [reply src/test/java/co/rsk/federate/signing/hsm/advanceBlockchain/HSMBookKeepingClientProviderTest.java:20]: Sure. And sorry the case I meant for class names is PascalCase, not camelCase, with the first letter of each separate word capitalized, no matter if is an acronym.

Take a look at `HsmBookkeepingClientImpl` class. You used PascalCase, but in other classes, you capitalized all the acronym`HSM`, for example for `HSMClientProtocol` class. So, my suggestion is to use only one style, this `Hsm` or this one `HSM`.

Take a look at this answer: https://stackoverflow.com/questions/2236807/java-naming-convention-with-acronyms

Comment (kelvinator07) [reply src/test/java/co/rsk/federate/signing/hsm/advanceBlockchain/HSMBookKeepingClientProviderTest.java:20]: `HSM` Is The Standard For HSM Related Classes, I'll Change That For `HsmBookkeepingClientImpl` But For Variable/Field Names, `hsmXXX` Is Fairly Okay. Thanks For The Update

**Review by nathanieliov — APPROVED** [review]

**Pull request merged**